launcher.py

```diff
@@ -0,0 +1,479 @@
+            sys.exit(1)
+
+    def _get_check_commands(self, system):
+        if system in self.check_cmd:
+            return self.check_cmd[system]
+        if "all" in self.check_cmd:
+            return self.check_cmd["all"]
+        return [f"{self.package_name.get(system, self.name)} --version"]
+
+    def _get_install_commands(self, system):
+        if system in self.install_cmd:
+            return self.install_cmd[system]
+        if "all" in self.install_cmd:
+            return self.install_cmd["all"]
+        if system not in self._installers:
+            logging.error(f"Unspported sytem {system}.")
+            sys.exit(1)
+
+        return self._installers[system]()
+
+    def _get_install_commands_linux(self):
+        import distro
+
+        distro = distro.id().lower()
+        if distro not in self._linux_distro_map:
+            logging.error(f"Unsupported linux distro {distro}.")
+            sys.exit(1)
+
+        return [f"{self._linux_distro_map[distro]} {self.package_name.get('Linux', self.name)}"]
+
+    def _get_install_commands_darwin(self):
+        return [f"brew install {self.package_name.get('Darwin', self.name)}"]
+
+    def _get_install_commands_windows(self):
+        if not check_msys2_installed():
+            install_msys2()
+        return [f"pacman -S --needed --noconfirm {self.package_name.get('Windows', self.name)}"]
+
+
+# pkg-config should come before sndfie and nlohmann-json
+dependencies = [
+    DependencyHandler("cmake", {"Windows": "mingw-w64-x86_64-cmake"}, {"all": ["cmake --version"]}),
+    DependencyHandler(
+        "g++",
+        {"Windows": "mingw-w64-x86_64-gcc", "Darwin": "gcc"},
+        {"all": ["g++ --version"]},
+        {"Windows": ["pacman -S --needed --noconfirm base-devel mingw-w64-x86_64-toolchain"]},
+    ),
+    DependencyHandler("pkg-config"),
+    DependencyHandler("ffmpeg", {"Windows": "mingw-w64-x86_64-ffmpeg"}, {"all": ["ffmpeg -version"]}),
+    DependencyHandler(
+        "libsndfile",
+        {"Windows": "mingw-w64-x86_64-libsndfile", "Linux": "libsndfile1-dev"},
+        {"all": ["pkg-config --exists sndfile"]},
+    ),
+    DependencyHandler(
+        "nlohmann-json",
+        {"Windows": "mingw-w64-x86_64-nlohmann-json", "Linux": "nlohmann-json3-dev"},
+        {"all": ["pkg-config --exists nlohmann_json"]},
+    ),
+]
+
+
+def check_msys2_installed():
+    try:
+        run_command("pacman --version")
+        return True
+    except (FileNotFoundError, subprocess.CalledProcessError):
+        return False
+
+
+def check_internet_connectivity():
+    """For Windows check internet connectivity"""
+    try:
+        logging.debug("Checking internet connectivity... ")
+        # Ping Google's public DNS server
+        run_command("ping -n 1 8.8.8.8")
+        logging.debug("Internet connectivity OK")
+    except subprocess.CalledProcessError:
+        logging.error("No internet connection detected.")
+        sys.exit(1)
```

Comment:
this should be cross platform and checked for all platforms. We rely on having libsndfile installed on the machine for instance.

Reply:
On linux systems package are not deleted permanently and reside on cache, so checking internet connection may not be important for linux. By the way we can check internet connection by just changing `-n` to `-c`.

Reply:
Sure, but what about a user that has a missing dependency? We should not assume the dependencies will be available by being cached or another means. In any case we should check for connectivity, and as you suggest, you can use `platform.system()` for example and make it generic.